client/matching/client.go

```diff
@@ -232,6 +241,49 @@ func (c *clientImpl) ListTaskListPartitions(
 	return client.ListTaskListPartitions(ctx, request, opts...)
 }
 
+func (c *clientImpl) GetTaskListsByDomain(
+	ctx context.Context,
+	request *types.GetTaskListsByDomainRequest,
+	opts ...yarpc.CallOption,
+) (*types.GetTaskListsByDomainResponse, error) {
+	opts = common.AggregateYarpcOptions(ctx, opts...)
+	clients, err := c.clientIterator()
+	if err != nil {
+		return nil, err
+	}
+
+	var futures []future.Future
+	for _, client := range clients {
+		future, settable := future.NewFuture()
+		go func() {
+			settable.Set(client.(Client).GetTaskListsByDomain(ctx, request, opts...))
```

Comment:
Do we need to make sure the ctx here has a finite/reasonable timeout here? Otherwise, the created goroutine may be blocked, especially when one future returns an error and we just return from the function without waiting for other futures?

Reply:
Looks like the loop variable (`client`) is being captured here. Will that be an issue? shall we pass in the client as a parameter to this inline func?

Reply:
updated